### PR TITLE
2554 Add channel dim if necessary

### LINKED ISSUE #2554: `AsDiscrete` needs to add channel dim for scalar label values
**Is your feature request related to a problem? Please describe.**
An question raised from the internal team:
Some PyTorch loss functions (like `torch.nn.CrossEntropyLoss`) require the label data has no channel dim, but postprocessing transforms and metrics require the label data with channel dim if multiple-classes. Need to add channel dim in post processing if ndim=0.

Actually, we had such feature in previous versions:
https://github.com/Project-MONAI/MONAI/blob/0.5.3/monai/transforms/post/array.py#L95

`VoteEnsemble` already considers the no-channel data:
https://github.com/Project-MONAI/MONAI/blob/0.6.0rc2/monai/transforms/post/array.py#L413

I think maybe we deleted it by mistake when developing the `decollate batch`.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2554 .

### Description
This PR added the channel dim if passing a scalar label tensor, also added the unit tests to cover it.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Just realized that `one_hot` utility already has a similar logic to add channel dim internally:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/utils.py#L74
Updated the PR to remove the redundant logic in One-Hot and keep the unit tests for it.

Thanks.

**Review by yiheng-wang-nv — APPROVED** [review]
